web/crux-ui/src/validations/auth.ts

```diff
@@ -1,22 +1,43 @@
 import { OidcProvider, OIDC_PROVIDER_VALUES } from '@app/models'
 import * as yup from 'yup'
-import { identityNameRule } from './common'
+import { identityNameRule, passwordLengthRule } from './common'
 
 export const registerWithPasswordSchema = yup.object().shape({
+  password: passwordLengthRule.required(),
   email: yup.string().email().required(),
-  password: yup.string().required(),
   firstName: identityNameRule.required(),
   lastName: identityNameRule.min(0),
 })
 
 export const registerWithOidcSchema = yup.object().shape({
   provider: yup.mixed<OidcProvider>().oneOf([...OIDC_PROVIDER_VALUES]),
+  firstName: identityNameRule.required(),
+  lastName: identityNameRule.min(0),
+  email: yup.string().email().required(),
 })
 
-export const registerSchema = yup.object().when('method', {
-  is: 'password',
-  then: () => registerWithPasswordSchema,
-  otherwise: () => registerWithOidcSchema,
+export const registerSchema = yup.object().shape({
```

Comment:
What was the problem with the original solution?

Reply:
It returned a yup object with empty fields so it didn't validate anything.

Reply:
yup can only validate fields which is in the schema and since 'method' is not part of the schema its value is always undefined.